modules/libcom/src/osi/osiSock.h

```diff
@@ -20,14 +20,43 @@
 #include "osdSock.h"
 #include "ellLib.h"
 
+//#define NETDEBUG
+
+//#ifndef EPICS_HAS_IPV6
+//#define EPICS_HAS_IPV6 1
+//#endif
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+#ifdef NETDEBUG
+#define EPICSSOCKET_PRINT_DEBUG_FLAG        (1<<0)
+#else
+#define EPICSSOCKET_PRINT_DEBUG_FLAG        (0)
+#endif
+#define EPICSSOCKET_ENABLEADDRESSREUSE_FLAG (1<<1)
+#define EPICSSOCKET_ENABLEBROADCASTS_FLAG   (1<<2)
+#define EPICSSOCKET_BIND_LOCALPORT_FLAG     (1<<3)
+#define EPICSSOCKET_CONNECT_IPV4            (1<<4)
+/* leave out 5 */
+#define EPICSSOCKET_CONNECT_IPV6            (1<<6)
+
+
 struct sockaddr;
 struct sockaddr_in;
 struct in_addr;
 
+typedef union osiSockAddr46 {
+    struct sockaddr_in  ia;
+    struct sockaddr     sa;
+#if EPICS_HAS_IPV6
```

Comment:
Try replacing `#if EPICS_HAS_IPV6` with `#ifdef AF_INET6`.  This should avoid the need for makefile games.